ui/helpers/location.js

```diff
@@ -5,3 +5,11 @@ export const getAllUrlParams = function getAllUrlParams() {
 export const getUrlParam = function getUrlParam(name) {
   return getAllUrlParams().get(name);
 };
+
+export const getAllUrlParamsAsObject = function getAllUrlParamsAsObject() {
```

Comment:
I think this is fine for now since it's pre-existing, but it seems like perhaps the callers of `getAllUrlParamsAsObject()` would be better off manipulating the `URLSearchParams` object directory rather than as a simple object, since they are adding values and then turning it back into a query string anyway?

Reply:
parseQueryParams in helpers/url.jsx is doing something similar to this.

Reply:
Ghah!  You're totally right.  Sheesh, with so many helpers, I lost track...  I'll move to using that instead.  Though the ``reduce`` is a better option for that function, so I'll update it.

Reply:
Thanks for all the great feedback on this, both of you.  It's an odd thing to go between the ``URLSearchParams`` object and simple ``Object``s.  But the simple objects are really great for supporting spread syntax.  It allows you to tersely overwrite params (or allow defaults to be overwritten).  And in Perfherder, it's calling to the ``PushModel`` with params that differ from the url, so it's a little shorter for them to be able to pass a plain object rather than constructing a ``URLSearchParams`` each time.

I'm not opposed to changing that, but it seemed like the simplest approach for now, IMHO.  :)